### ISSUE TITLE
IPython console error in Spyder 3.0.0b3

### ISSUE THREAD

**Issue by iamsc** [spyder-ide/spyder#3212]
## Description of your problem

When I want to call any function and type the left bracket in ipython console, the error message will prompt in internal console. The error message is as below:

```
Traceback (most recent call last):
  File "/usr/local/opt/pyenv/versions/3.5.1/lib/python3.5/site-packages/qtconsole/base_frontend_mixin.py", line 163, in _dispatch
    handler(msg)
  File "/usr/local/opt/pyenv/versions/3.5.1/lib/python3.5/site-packages/spyderlib/widgets/ipython.py", line 377, in _handle_inspect_reply
    signature = self.get_signature(content)
  File "/usr/local/opt/pyenv/versions/3.5.1/lib/python3.5/site-packages/spyderlib/widgets/ipython.py", line 296, in get_signature
    text = ANSI_OR_SPECIAL_PATTERN.sub('', text)
TypeError: expected string or bytes-like object
```

**What steps will reproduce the problem?**

1. For example, we try to call the `print('test')`.
2. To input the `print(`.
3. the error message will be prompt in the internal console.

How could I solve this problem?

## Versions and main components

* Spyder Version: Spyder 3.0.0b3 
* Python Version: 3.5.1 64bits (pyenv)
* Operating system: Mac 10.11.5

## Dependencies

jedi >=0.8.1    :  0.9.0 (OK)
matplotlib >=1.0:  1.5.1 (OK)
nbconvert >=4.0 :  4.2.0 (OK)
numpy >=1.7     :  1.11.0 (OK)
pandas >=0.13.1 :  0.18.1 (OK)
pep8 >=0.6      :  1.7.0 (OK)
psutil >=0.3    :  4.2.0 (OK)
pyflakes >=0.6.0:  1.2.3 (OK)
pygments >=1.6  :  2.1.3 (OK)
pylint >=0.25   :  1.5.6 (OK)
qtconsole >=4.0 :  4.2.1 (OK)
rope >=0.9.4    :  0.9.4-1 (OK)
sphinx >=0.6.6  :  1.4.3 (OK)
sympy >=0.7.3   :  1.0 (OK)

**Comment by stonebig** [comment]
same thing on windows / Python3.5.1

**Comment by ccordoba12** [comment]
Thanks for the confirmation @stonebig. I'll take a look at it for beta4 :-)

**Comment by stonebig** [comment]
sames with Python3.4 / PyQt5.5

**Comment by stonebig** [comment]
this is a `<class 'dict'>`

**Comment by stonebig** [comment]
a basic fix may be this:
 - replace File "/usr/local/opt/pyenv/versions/3.5.1/lib/python3.5/site-packages/spyderlib/widgets/ipython.py", line 296, in get_signature

````
            text = ANSI_OR_SPECIAL_PATTERN.sub('', text)
````
per
````
            if type(text) == type({'e':5}):
                text= text['text/plain']
            text = ANSI_OR_SPECIAL_PATTERN.sub('', text)
````

**Comment by ccordoba12** [comment]
I can't reproduce this problem, and the proposed solution seems a bit fishy to me.

@stonebig, what do you get when you print the `data` variable that is one line above the `text` one?

**Comment by stonebig** [comment]
apparently this:
````
`{'text/plain': {'text/html': "<h1>Docstring</h1>\n<pre>print(value, ..., sep=' ', end='\\n', file=sys.stdout, flush=False)\n\nPrints the values to a stream, or to sys.stdout by default.\nOptional keyword arguments:\nfile:  a file-like object (stream); defaults to the current sys.stdout.\nsep:   string inserted between values, default a space.\nend:   string appended after the last value, default a newline.\nflush: whether to forcibly flush the stream.</pre>\n<h1>Type</h1>\n<pre>builtin_function_or_method</pre>\n", 'text/plain': "\x1b[1;31mDocstring\x1b[0m:\nprint(value, ..., sep=' ', end='\\n', file=sys.stdout, flush=False)\n\nPrints the values to a stream, or to sys.stdout by default.\nOptional keyword arguments:\nfile:  a file-like object (stream); defaults to the current sys.stdout.\nsep:   string inserted between values, default a space.\nend:   string appended after the last value, default a newline.\nflush: whether to forcibly flush the stream.\n\x1b[1;31mType\x1b[0m:\nbuiltin_function_or_method\n"}}
````

**Comment by ccordoba12** [comment]
I see. This is a change brought by IPython 5.0 beta2 (that's why I was unable to reproduce it). Then your PR is the right one @stonebig :-)

**Comment by Carreau** [comment]
Sorry about that it was a mistake on my part I was to fast to review a change in beta2. 
Beta 3 should revert the behavior (IIUC).

**Comment by ccordoba12** [comment]
@Carreau, thanks for letting us know about it. So things will be the same as in IPython 4.0?

**Comment by Carreau** [comment]
As far as I can tell this was introduced in https://github.com/ipython/ipython/pull/9551 and fixed in https://github.com/ipython/ipython/pull/9576. The behavior should be the same as 4.0, at least for this. 

We have some changes, but most of them affect the console UI, not the APIs (too much). 

And generally if you see a real change of API where the same function returning different object on two major versions without a big warning and explanation feel free to open an issue, it is likely a mistake on our part. Worst case it is not and we should document it more. Especially if the change happened in between beta releases it is likely a mistake.

**Comment by ccordoba12** [comment]
Ok, will report things with you guys if things break like this from now on :-)

I'll close this issue once I have a chance to test IPython 5.0 beta3.

**Comment by ccordoba12** [comment]
I can confirm this problem was fixed in IPython 5.0 beta3.

@iamsc, please update with this command:

    pip install --pre -U ipython

**Comment by iamsc** [comment]
It works!!! Thanks~
